### PR TITLE
error handling, keeping in redux

### PR DESCRIPTION
**Pull request by rafonersisyan**

### CODE REVIEW

File: components/errorPopup/ErrorPopup.tsx [hunk 1]
```diff
@@ -1,32 +1,42 @@
...
+import { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { errorHandling } from "./errorPopup.slice";
+import { document, mainWrapper, iconWrapper,
+  textWrapper, titleStyle, subTextStyle } from "./style";
 
 export const ErrorPopup = () => {
-  const [ isShown, setIsShown ] = useState(true);
+  const dispatch = useDispatch();
+  const data = useSelector((state: any) => state.error.error);
```
Comment (hrachocode) [inline components/errorPopup/ErrorPopup.tsx:11]:
```suggestion
  const data = useSelector((state: IRootState) => state.error.error);
```

File: components/errorPopup/ErrorPopup.tsx [hunk 2]
```diff
@@ -1,32 +1,42 @@
...
+  const [ isShown, setIsShown ] = useState(false);
+
+  useEffect(() => {
+    setTimeout(() => {
```
Comment (hrachocode) [inline components/errorPopup/ErrorPopup.tsx:15]: why you need set timeout here?

File: components/errorPopup/errorPopup.slice.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { createSlice } from "@reduxjs/toolkit";
+import { ERROR } from "constants/reducers";
+
+export const errorPopupSlice = createSlice({
+  name: ERROR,
+  initialState: { error: { isError: false } },
```
Comment (hrachocode) [inline components/errorPopup/errorPopup.slice.ts:6]:
```suggestion
  initialState: { 
      isError: false 
  },
```

?

File: components/errorPopup/style.ts [hunk 4]
```diff
@@ -1,10 +1,21 @@
+export const document = {
+  width: "100%",
+  height: "90vh",
```
Comment (hrachocode) [inline components/errorPopup/style.ts:3]: will this work for larger screens? you can minimize the screen with `cntrl` + `-`

to check not only responsive, but also larger screens

File: pages/about-us.tsx [hunk 5]
```diff
@@ -27,12 +28,18 @@ interface IAboutUs {
...
   const dispatch = useDispatch();
 
   useEffect(() => {
+    if(error){
+      dispatch(errorHandling({ isError: true }));
+    }else {
+      dispatch(errorHandling({ isError: false }));
+    }
     dispatch(changeData(texts));
   });
```
Comment (hrachocode) [inline pages/about-us.tsx:44]:
```suggestion
    if(error){
      dispatch(errorHandling({ isError: true }));
    }else {
      dispatch(errorHandling({ isError: false }));
    }
    dispatch(changeData(texts));
  });
```

create a separate hook called useFetchedData(error, data) and move all this logic there and use overall in project.

File: pages/join-us.tsx [hunk 6]
```diff
@@ -41,15 +48,25 @@ const JoinUs = ({ texts, positions }: IJoinUs) => {
...
+        texts: formatApiTexts(texts),
+        positions: positions || []
+      }
+    };
+  } catch (error: any) {
+    return {
+      props: {
+        texts: [],
+        positions: [],
+        error: error?.data.message
```
Comment (hrachocode) [inline pages/join-us.tsx:65]:
```suggestion
        error: error?.data?.message
```

same everywhere

**Review by hrachocode — CHANGES_REQUESTED** [review]

File: pages/about-us.tsx (hunk 5 above)
Comment (rafonersisyan) [reply pages/about-us.tsx:44]: i can't use custom hooks in useEffect

File: components/errorPopup/style.ts (hunk 4 above)
Comment (rafonersisyan) [reply components/errorPopup/style.ts:3]: yes

File: components/errorPopup/ErrorPopup.tsx (hunk 2 above)
Comment (rafonersisyan) [reply components/errorPopup/ErrorPopup.tsx:15]: removed

**Review by hrachocode — APPROVED** [review]

**Pull request merged**